internal/bminventory/inventory_v2_handlers.go

```diff
@@ -1039,6 +1043,10 @@ func kernelArgsAppendStr(infraEnv *common.InfraEnv) (string, error) {
 }
 
 func (b *bareMetalInventory) bootIPXEScript(ctx context.Context, infraEnv *common.InfraEnv) (string, error) {
+	if !b.isImageServiceEnabled() {
+		return "", errors.New("image service is disabled")
```

Comment:
This should return specifically an `ApiError`, otherwise it will get the default (500)